openedx/features/marketplace/views.py

```diff
@@ -1,24 +1,145 @@
+from django.db.models import Q
 from django.shortcuts import get_object_or_404
 from django.urls import reverse
 from django.utils.decorators import method_decorator
 from django.views import View, generic
+from django.views.generic.list import ListView
+from django_countries import countries
 
 from edxmako.shortcuts import render_to_response
 from openedx.features.marketplace.decorators import has_affiliated_user
 from openedx.features.marketplace.forms import MarketplaceRequestForm
 
+from .constants import (
+    DJANGO_COUNTRIES_KEY_INDEX,
+    DJANGO_COUNTRIES_VALUE_INDEX,
+    MARKETPLACE_PARAM_CITY_KEY,
+    MARKETPLACE_PARAM_COUNTRY_KEY,
+    MARKETPLACE_PARAM_QUERY_KEY,
+    ORGANIZATIONAL_PROBLEM_CHOICES_VIEW,
+    ORGANIZATIONAL_PROBLEM_TYPE_DELIVERY_SERVICES_KEY,
+    ORGANIZATIONAL_PROBLEM_TYPE_FUNDING_KEY,
+    ORGANIZATIONAL_PROBLEM_TYPE_HEALTH_CARE_SUPPLIES_KEY,
+    ORGANIZATIONAL_PROBLEM_TYPE_HUMAN_RESOURCES_KEY,
+    ORGANIZATIONAL_PROBLEM_TYPE_MENTORSHIP_KEY,
+    ORGANIZATIONAL_PROBLEM_TYPE_ONLINE_TRAINING_KEY,
+    ORGANIZATIONAL_PROBLEM_TYPE_OTHER_KEY,
+    ORGANIZATIONAL_PROBLEM_TYPE_REMOTE_WORKING_TOOLS_KEY,
+    ORGANIZATIONAL_SECTOR_ART_AND_CULTURE,
+    ORGANIZATIONAL_SECTOR_CLEAN_ENERGY,
+    ORGANIZATIONAL_SECTOR_CLIMATE_CHANGES,
+    ORGANIZATIONAL_SECTOR_EDUCATION,
+    ORGANIZATIONAL_SECTOR_ENVIRONMENTAL_CONSERVATION,
+    ORGANIZATIONAL_SECTOR_GENDER_EQUALITY,
+    ORGANIZATIONAL_SECTOR_HEALTH_AND_WELL_BEING,
+    ORGANIZATIONAL_SECTOR_HUMAN_RIGHTS,
+    ORGANIZATIONAL_SECTOR_OTHER_KEY,
+    ORGANIZATIONAL_SECTOR_SANITATION,
+    ORGANIZATIONAL_SECTOR_SOCIAL_JUSTICE,
+    ORGANIZATIONAL_SECTOR_WORK_AND_ECONOMIC_GROWTH,
+    ORGANIZATION_SECTOR_CHOICES_VIEW
+)
 from .models import MarketplaceRequest
 
 
-class MarketplaceListingView(View):
+class MarketplaceListView(ListView):
+    model = MarketplaceRequest
+    context_object_name = 'marketplace_list'
+    paginate_by = 10
     template_name = 'features/marketplace/marketplace_listing.html'
+    ordering = ['-created']
+    template_engine = 'mako'
 
-    def get(self, request, *args, **kwargs):
-        return render_to_response(self.template_name, {})
+    def get_checked_choices(self, choices):
+        """
+        This method takes list of tuples (choices) and returns a list
+        of choice keys that exist in self.request.GET
+        :param: choices
+        :return: list of choice keys that are also in self.request.GET
+        """
+        params = self.request.GET
+        tuple_key_index = 0
+        return [choice[tuple_key_index] for choice in choices if params.get(choice[tuple_key_index])]
+
+    def get_queryset(self):
+        params = self.request.GET
+
+        query = params.get(MARKETPLACE_PARAM_QUERY_KEY)
+        city = params.get(MARKETPLACE_PARAM_CITY_KEY)
+        country = params.get(MARKETPLACE_PARAM_COUNTRY_KEY)
+
+        request_type_list = self.get_checked_choices(ORGANIZATIONAL_PROBLEM_CHOICES_VIEW)
```

Comment:
What this `request_type_list` actually is. Can we give it more appropriate name?